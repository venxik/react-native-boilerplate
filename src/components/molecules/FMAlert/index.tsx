import React, { memo } from 'react';

import { Alert, Box, CloseIcon, HStack, IconButton, Text, VStack } from 'native-base';

import type { IAlertProps } from 'native-base';

export type IFMAlertProps = {
  title?: string;
  description?: string;
} & IAlertProps;

export const FMAlert = memo((props: IFMAlertProps) => {
  const { testID, title, description, ...baseProps } = props;

  return (
    <Alert testID={testID} {...baseProps}>
      <VStack space={2} flexShrink={1} w="100%">
        <HStack flexShrink={1} space={2} alignItems="center" justifyContent="space-between">
          <HStack flexShrink={1} space={2} alignItems="center">
            <Alert.Icon />
            <Text fontSize="md" fontWeight="medium" color="coolGray.800">
              {title}
            </Text>
          </HStack>
          <IconButton
            variant="unstyled"
            _focus={{
              borderWidth: 0,
            }}
            icon={<CloseIcon size="3" color="coolGray.600" />}
          />
        </HStack>
        <Box
          pl="6"
          _text={{
            color: 'coolGray.600',
          }}
        >
          {description}
        </Box>
      </VStack>
    </Alert>
  );
});
FMAlert.displayName = 'FAMlert';
