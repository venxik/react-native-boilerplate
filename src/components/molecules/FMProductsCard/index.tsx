import React from 'react';
import { Text, Box, Image, Pressable, HStack } from 'native-base';
import { heightPercentageToDP } from '@utils';

type ProductsCardProps = {
  image: string;
  name: string;
  desc: string;
  onPress?: () => void;
};

export default function FMProductsCard({ desc, image, name, onPress }: ProductsCardProps) {
  const heightBox = heightPercentageToDP('10');
  return (
    <Pressable
      onPress={onPress}
      borderWidth="1"
      mb={'4'}
      borderRadius={'xl'}
      p={'4'}
      testID={'FMProductsCard'}
    >
      <Box height={heightBox} borderRadius="xl" mb={4}>
        <HStack>
          <Image
            width={heightBox}
            height={heightBox}
            resizeMode="contain"
            source={{ uri: image }}
            alt="image"
          />
          <Box justifyContent="center" pl={5} flex={1}>
            <Text color="manta.100" fontFamily="mono" fontWeight={800} fontSize="xs">
              {name}
            </Text>
            <Text
              color="seaflor.100"
              fontFamily="heading"
              fontWeight={600}
              fontSize="md"
              ellipsizeMode="tail"
              numberOfLines={2}
            >
              {desc}
            </Text>
          </Box>
        </HStack>
      </Box>
    </Pressable>
  );
}
