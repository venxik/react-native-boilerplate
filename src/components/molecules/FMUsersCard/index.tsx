import React from 'react';
import { Text, Box, Image, Pressable, HStack } from 'native-base';
import { heightPercentageToDP } from '@utils';

type UsersCardProps = {
  image: string;
  name: string;
  desc: string;
  onPress?: () => void;
};

export default function FMUsersCard({ desc, image, name, onPress }: UsersCardProps) {
  const heightBox = heightPercentageToDP('10');
  return (
    <Pressable
      onPress={onPress}
      borderWidth="1"
      mb={'4'}
      borderRadius={'xl'}
      testID={'FMUsersCard'}
    >
      <Box height={heightBox} borderRadius="xl" mb={4}>
        <HStack>
          <Image
            width={heightBox}
            height={heightBox}
            resizeMode="contain"
            source={{ uri: image }}
            alt="image"
          />
          <Box justifyContent="center" pl={5}>
            <Text color="manta.100" fontFamily="mono" fontWeight={800} fontSize="xs">
              {name}
            </Text>
            <Text color="seaflor.100" fontFamily="heading" fontWeight={600} fontSize="md">
              {desc}
            </Text>
          </Box>
        </HStack>
      </Box>
    </Pressable>
  );
}
