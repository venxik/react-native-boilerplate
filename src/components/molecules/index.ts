import FMUsersCard from './FMUsersCard';
import FMProductsCard from './FMProductsCard';
export * from './FMAlert';

export { FMUsersCard, FMProductsCard };
