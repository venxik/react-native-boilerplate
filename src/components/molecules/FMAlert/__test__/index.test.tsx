import React from 'react';

import { FMAlert } from '@components/molecules';
import { render } from '@mocks';

describe('FMAlert', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(<FMAlert />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with children prop', () => {
    const container = render(
      <FMAlert testID="alert-example" variant={'solid'} title={'Selection successfully moved!'} />,
    );

    expect(container).toBeTruthy();
  });
});
