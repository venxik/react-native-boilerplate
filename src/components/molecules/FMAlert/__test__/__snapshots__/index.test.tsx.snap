// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FMAlert should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    space={3}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#bae6fd",
        "borderRadius": 4,
        "justifyContent": "flex-start",
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "flexShrink": 1,
          "width": "100%",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexShrink": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexShrink": 1,
            }
          }
        >
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <RNSVGSvgView
              accessibilityRole="image"
              align="xMidYMid"
              bbHeight={16}
              bbWidth={16}
              color={-16553567}
              dataSet={Object {}}
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              stroke=""
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "color": "#0369a1",
                    "height": 16,
                    "width": 16,
                  },
                  Object {
                    "flex": 0,
                    "height": 16,
                    "width": 16,
                  },
                ]
              }
              tintColor={-16553567}
              vbHeight={24}
              vbWidth={24}
            >
              <RNSVGGroup
                propList={
                  Array [
                    "stroke",
                  ]
                }
                stroke={null}
              >
                <RNSVGGroup>
                  <RNSVGPath
                    d="M12 0C9.62663 0 7.30655 0.703788 5.33316 2.02236C3.35977 3.34094 1.8217 5.21509 0.913451 7.4078C0.00519948 9.60051 -0.232441 12.0133 0.230582 14.3411C0.693605 16.6689 1.83649 18.807 3.51472 20.4853C5.19295 22.1635 7.33115 23.3064 9.65892 23.7694C11.9867 24.2324 14.3995 23.9948 16.5922 23.0865C18.7849 22.1783 20.6591 20.6402 21.9776 18.6668C23.2962 16.6935 24 14.3734 24 12C23.9966 8.81846 22.7312 5.76821 20.4815 3.51852C18.2318 1.26883 15.1815 0.00344108 12 0ZM12.25 5C12.5467 5 12.8367 5.08797 13.0834 5.25279C13.33 5.41762 13.5223 5.65189 13.6358 5.92597C13.7494 6.20006 13.7791 6.50166 13.7212 6.79263C13.6633 7.08361 13.5204 7.35088 13.3107 7.56066C13.1009 7.77044 12.8336 7.9133 12.5426 7.97118C12.2517 8.02905 11.9501 7.99935 11.676 7.88582C11.4019 7.77229 11.1676 7.58003 11.0028 7.33335C10.838 7.08668 10.75 6.79667 10.75 6.5C10.75 6.10217 10.908 5.72064 11.1893 5.43934C11.4706 5.15803 11.8522 5 12.25 5ZM14.5 18.5H10.5C10.2348 18.5 9.98043 18.3946 9.7929 18.2071C9.60536 18.0196 9.5 17.7652 9.5 17.5C9.5 17.2348 9.60536 16.9804 9.7929 16.7929C9.98043 16.6054 10.2348 16.5 10.5 16.5H11.25C11.3163 16.5 11.3799 16.4737 11.4268 16.4268C11.4737 16.3799 11.5 16.3163 11.5 16.25V11.75C11.5 11.6837 11.4737 11.6201 11.4268 11.5732C11.3799 11.5263 11.3163 11.5 11.25 11.5H10.5C10.2348 11.5 9.98043 11.3946 9.7929 11.2071C9.60536 11.0196 9.5 10.7652 9.5 10.5C9.5 10.2348 9.60536 9.98043 9.7929 9.79289C9.98043 9.60536 10.2348 9.5 10.5 9.5H11.5C12.0304 9.5 12.5391 9.71071 12.9142 10.0858C13.2893 10.4609 13.5 10.9696 13.5 11.5V16.25C13.5 16.3163 13.5263 16.3799 13.5732 16.4268C13.6201 16.4737 13.6837 16.5 13.75 16.5H14.5C14.7652 16.5 15.0196 16.6054 15.2071 16.7929C15.3946 16.9804 15.5 17.2348 15.5 17.5C15.5 17.7652 15.3946 18.0196 15.2071 18.2071C15.0196 18.3946 14.7652 18.5 14.5 18.5Z"
                    fill={
                      Array [
                        2,
                      ]
                    }
                    propList={
                      Array [
                        "fill",
                        "stroke",
                      ]
                    }
                    stroke={null}
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "width": 8,
              }
            }
          />
          <Text
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#1f2937",
                "fontFamily": undefined,
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "500",
                "letterSpacing": 0,
                "lineHeight": 24,
                "textDecorationLine": undefined,
              }
            }
          />
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "width": 8,
            }
          }
        />
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <RNSVGSvgView
            accessibilityRole="image"
            align="xMidYMid"
            bbHeight={12}
            bbWidth={12}
            color={-11840157}
            dataSet={Object {}}
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke=""
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "color": "#4b5563",
                  "height": 12,
                  "width": 12,
                },
                Object {
                  "flex": 0,
                  "height": 12,
                  "width": 12,
                },
              ]
            }
            tintColor={-11840157}
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              propList={
                Array [
                  "stroke",
                ]
              }
              stroke={null}
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M12 9.77778L4.22222 2L2 4.22222L9.77778 12L2 19.7778L4.22222 22L12 14.2222L19.7778 22L22 19.7778L14.2222 12L22 4.22222L19.7778 2L12 9.77778Z"
                  fill={
                    Array [
                      2,
                    ]
                  }
                  propList={
                    Array [
                      "fill",
                      "stroke",
                    ]
                  }
                  stroke={null}
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 8,
          }
        }
      />
      <View
        dataSet={Object {}}
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
